tool/tctl/common/auth_command.go

```diff
@@ -713,10 +713,11 @@ func (a *AuthCommand) generateUserKeys(ctx context.Context, clusterAPI auth.Clie
 		return nil
 	}
 	if reqExpiry.Sub(expires) > time.Minute {
-		log.Warnf("Requested TTL of %s was not granted. User may have a role with a shorter max session TTL"+
-			" or an existing session ending before the requested TTL. Proceeding with %s",
+		return trace.Wrap(fmt.Errorf(`Error: the credential was not issued because the requested TTL exceeded the maximum allowable 
```

Comment:
You can use `trace.Errorf` or `trace.AccessDenied` here instead of wrapping an error produced by `fmt.Errorf`.

Additionally, please remove the newlines and tab characters from the error message.

Reply:
Fixed on https://github.com/gravitational/teleport/pull/15384/commits/d4f1241b5fa0d633add42a4d5c817fc3c7ba0c11, let me know if this PR needs additional changes!